pymongo/mongo_client.py

```diff
@@ -1405,6 +1408,10 @@ def is_retrying():
                 # Add the RetryableWriteError label, if applicable.
                 _add_retryable_write_error(exc, max_wire_version)
                 retryable_error = exc.has_error_label("RetryableWriteError")
+                if self._indefinite_error and exc.has_error_label("NoWritesPerformed"):
+                    raise self._indefinite_error from exc
+                if isinstance(exc, WriteConcernError):
+                    self._indefinite_error = exc
```

Comment:
_indefinite_error  is not needed. `last_error` already tracks the error that should be propagated.

Reply:
Why do you think we need to track the last WriteConcernError? This change should not track WriteConcernErrors any differently than other errors. This change is about errors labeled with NoWritesPerformed which may or may not be a WriteConcernError.